### PR TITLE
#194 Fix LinkedIn parsing id.

### PR DESCRIPTION
**Pull request by eg04lt3r**
Fix Linkedin parsing id and implement test case for class. #194 - original isuue.

### CODE REVIEW

**Comment by davvd** [comment]
@eg04lt3r Thanks, I will find someone to review this PR soon

**Comment by davvd** [comment]
@pinaf please review,thanks

**Comment by pinaf** [comment]
@davvd @yegor256 is this valid? I reviewed #123 for this same issue, which is still open. Now issue was reassigned and there's another PR + review going on.. @yegor256 please advise.

**Comment by yegor256** [comment]
@pinaf we should review each pull request individually. as a reviewer you shouldn't really care what issue this pull request is solving. everything should be clear from the branch. look at the branch and make a decision - does it improve the code base? does it follow the design idea of the product? If yes, approve it.

File: src/main/java/org/takes/http/BkTimeable.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.net.Socket;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Back decorator with maximum lifetime.
+ * <p>The class is immutable and thread-safe.
+ * @author Aleksey Kurochka (eg04lt3r@gmail.com)
+ * @version $Id$
+ */
```
Comment (pinaf) [inline src/main/java/org/takes/http/BkTimeable.java]: @eg04lt3r `@since 0.15`

File: src/main/java/org/takes/http/BkTimeable.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        this.origin = back;
+        this.latency = ltc;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    public void accept(final Socket socket) throws IOException {
+        this.origin.accept(socket);
+        final Thread callerThread = Thread.currentThread();
+        final Thread monitor = new Thread(
```
Comment (pinaf) [inline src/main/java/org/takes/http/BkTimeable.java]: @eg04lt3r we've talked about this approach before in #123

**Comment by pinaf** [comment]
@eg04lt3r I have a feeling some commits don't belong here. this code looks like it is mixed up. maybe all you wanted to push was the linkedin patch?

**Comment by pinaf** [comment]
@eg04lt3r 3 comments above. something is wrong with this PR.

**Comment by eg04lt3r** [comment]
@pinaf Sorry, i deleted wrong code from this pull request. Please, review last commit.

**Comment by pinaf** [comment]
@eg04lt3r no problem

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@eg04lt3r how about a test first?

**Comment by eg04lt3r** [comment]
@yegor256 Review last commit. I implemented test according to puzzle.

**Comment by yegor256** [comment]
@eg04lt3r mocking is a rather dangerous technique (http://www.yegor256.com/2014/09/23/built-in-fake-objects.html), but this wire mocking or power mocking is a pure evil. Having this unit test in place you basically prohibit us to change the class in the future. If we want to refactor the class and start using Apache Http Client instead of jcabi-http - we won't be able to do this. We'll have to throw away the unit test first.

**Comment by eg04lt3r** [comment]
@yegor256  according to puzlle of @dmzaytsev I implemented unit test for PsLinkedIn. It's good solution for similar tests in the future. And I do not see any difficulties with using it. Also, if PsLinkedin will use Apache instead of Jcabi no reason to rewrite test.

**Comment by yegor256** [comment]
@eg04lt3r ah, I got it, sorry about the last comment, I was wrong. But I think it would be better to use jcabi-http mocking, since we're using the library already, see: http://www.yegor256.com/2014/04/18/jcabi-http-server-mocking.html

**Comment by eg04lt3r** [comment]
@yegor256 Ok. No problem. Does it track additional time of this implementation?

**Comment by yegor256** [comment]
@eg04lt3r see http://www.teamed.io/policy.html#5 and http://www.teamed.io/policy.html#6

**Comment by pinaf** [comment]
@eg04lt3r are you still working on this one? If not, please close the PR.

**Comment by eg04lt3r** [comment]
@pinaf I'll commit today.

**Comment by eg04lt3r** [comment]
@yegor256  After investigation of jcabi-http, I should say that more flexible way to test similar cases is to use WireMock. If use jcabi I should refactor PsLinkedin constructor to use JdkRequest, what obliges   us to rework class if we decide use Apache Http Client.

**Comment by yegor256** [comment]
@eg04lt3r you can use Takes itself, like it's done in https://github.com/yegor256/takes/pull/228/files

**Comment by eg04lt3r** [comment]
@yegor256 I reworked it, and remove WireMock dependency. Please, review.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@davvd Quality is good here.

**Comment by davvd** [comment]
> @davvd Quality is good here.

@elenavolokhova thank you

**Comment by davvd** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 57125656... done, I added **21 mins** in payment `AP-34575035VX696643H`, 383 hours and 47 mins spent to complete this... review comments (c=6) added as a bonus... +21 added to your rating, at the moment it is: [+7809](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**